src/tools/wasm-split/instrumenter.cpp

```diff
@@ -179,7 +179,18 @@ void Instrumenter::addProfileExport() {
     }
   }
 
-  // TODO: export the memory if it is not already exported.
+  // Export the memory if it is not already exported.
+  bool memoryExported = false;
+  for (auto& ex : wasm->exports) {
+    if (ex->kind == ExternalKind::Memory) {
+      memoryExported = true;
+      break;
+    }
+  }
+  if (!memoryExported) {
+    wasm->addExport(Builder::makeExport(
+      "profile-memory", wasm->memory.name, ExternalKind::Memory));
```

Comment:
It would be good to use `Names::getValidExportName` here. Also, what do you think about just calling the export "memory" by default? That seems simpler, and there probably won't already be a "memory" export for that to clash with.

Reply:
Updated to use `Names`, good point.

I actually picked `profile-memory` intentionally because it seems more future-proof. Once we have multiple memories, we'd want to use a separate memory here I think, so being clear on the purpose of the memory seems good.